app/uk/gov/hmrc/cgtpropertydisposals/service/UpscanService.scala

```diff
@@ -0,0 +1,151 @@
+
+  def getUpscanSnapshot(cgtReference: CgtReference): EitherT[Future, Error, UpscanSnapshot]
+
+  def storeFileDescriptorData(fd: UpscanFileDescriptor): EitherT[Future, Error, Unit]
+
+  def storeCallBackData(cb: UpscanCallBack): EitherT[Future, Error, Unit]
+
+  def getFileDescriptor(
+    cgtReference: CgtReference,
+    fileDescriptorId: FileDescriptorId
+  ): EitherT[Future, Error, Option[UpscanFileDescriptor]]
+
+  def getUpscanFileDescriptor(fileDescriptorId: FileDescriptorId): EitherT[Future, Error, Option[UpscanFileDescriptor]]
+
+  def updateUpscanFileDescriptorStatus(upscanFileDescriptor: UpscanFileDescriptor): EitherT[Future, Error, Boolean]
+
+  def getAllUpscanCallBacks(cgtReference: CgtReference): EitherT[Future, Error, List[UpscanCallBack]]
+
+  def downloadFilesFromS3(
+    snapshot: UpscanSnapshot,
+    urls: List[UpscanCallBack]
+  ): EitherT[Future, Error, List[Either[Error, FileAttachment]]]
+}
+
+@Singleton
+class UpscanServiceImpl @Inject() (
+  upscanConnector: UpscanConnector,
+  upscanFileDescriptorRepository: UpscanFileDescriptorRepository,
+  upscanCallBackRepository: UpscanCallBackRepository,
+  configuration: Configuration
+)(implicit executionContext: ExecutionContext)
+    extends UpscanService
+    with Logging {
+
+  private def getUpscanInitiateConfig[A: Configs](key: String): A =
+    configuration.underlying
+      .get[A](s"microservice.services.upscan-initiate.$key")
+      .value
+
+  private val s3UrlExpiryTime: Long = getUpscanInitiateConfig[Long]("s3url.expiry-time")
+
+  override def getUpscanSnapshot(
+    cgtReference: CgtReference
+  ): EitherT[Future, Error, UpscanSnapshot] =
+    for {
+      fds <- upscanFileDescriptorRepository.getAll(cgtReference)
+      d   <- EitherT.fromEither[Future](computeUpscanSnapshot(fds))
+    } yield d
+
+  override def updateUpscanFileDescriptorStatus(
+    upscanFileDescriptor: UpscanFileDescriptor
+  ): EitherT[Future, Error, Boolean] =
+    upscanFileDescriptorRepository.updateUpscanUploadStatus(upscanFileDescriptor)
+
+  def getUpscanFileDescriptor(
+    fileDescriptorId: FileDescriptorId
+  ): EitherT[Future, Error, Option[UpscanFileDescriptor]] =
+    upscanFileDescriptorRepository.get(fileDescriptorId)
+
+  override def storeFileDescriptorData(fd: UpscanFileDescriptor): EitherT[Future, Error, Unit] =
+    upscanFileDescriptorRepository.insert(fd)
+
+  override def storeCallBackData(cb: UpscanCallBack): EitherT[Future, Error, Unit] =
+    upscanCallBackRepository.insert(cb)
+
+  override def getFileDescriptor(
+    cgtReference: CgtReference,
+    fileDescriptorId: FileDescriptorId
+  ): EitherT[Future, Error, Option[UpscanFileDescriptor]] =
+    upscanFileDescriptorRepository.get(fileDescriptorId)
+
+  override def downloadFilesFromS3(
+    snapshot: UpscanSnapshot,
+    urls: List[UpscanCallBack]
+  ): EitherT[Future, Error, List[Either[Error, FileAttachment]]] = {
+    implicit val cs: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
```

Comment:
could you not use the injected execution context?

Reply:
`cats io` dep will be remvoed

Reply:
fixed